File: Tutor Demos/Week 5/Account.java
import java.text.*;

public class Account {
    private AccountType type;
    private double balance;

    public Account(AccountType type) {
        this.type = type;
        balance = readBalance();
    }

    private double readBalance() {
        System.out.print("Initial " + type + " balance: $");
        return In.nextDouble();
    }

    public boolean has(double amount) {
        return balance >= amount;
    }

    public void deposit(double amount) {
        balance += amount;
    }

    public void withdraw(double amount) {
        balance -= amount;
    }

    public void transferTo(double amount, Account target) {
        balance -= amount;
        target.balance += amount;
    }

    @Override
    public String toString() {
        return type + " account has $" + formatted(balance);
    }

    private String formatted(double amount) {
        return new DecimalFormat("###,##0.00").format(amount);
    }
}

File: Tutor Demos/Week 5/Python/account.py
class Account:
    def __init__(self, type):
        self.type = type
        self.balance = float(input(f"Initial {self.type} balance: $"))

    def has(self, amount):
        return self.balance >= amount

    def deposit(self, amount):
        self.balance += amount

    def withdraw(self, amount):
        self.balance -= amount

    def transfer_to(self, amount, target):
        self.balance += amount
        target.withdraw(amount)

    def __str__(self):
        return f"{self.type} account has ${self.balance:.2f}"
